### PR TITLE
Issue 1763 expired keys recognition

### PR DESCRIPTION
**Pull request by michael-volynets**
Issue #1763 
Time Spent: 7h 

Went smoothly

something happened with test.ci.compose Gmail account (the test fails on `master` branch too).

On `/api/message/token` the Flowcrypt API returns 401 error (but response status code is 200) and front-end shows this dialog with this text: `Your FlowCrypt account information is outdated, please review your account settings`, and the test fails on that step

### CODE REVIEW

File: extension/chrome/elements/pgp_pubkey.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ Catch.try(async () => {
   const minimized = uncheckedUrlParams.minimized === true;
 
   const { keys: pubs } = await openpgp.key.readArmored(armoredPubkey);
+  const isUsableButExpired = await Pgp.key.usableButExpired(pubs[0]) || undefined;
```
Comment (tomholub) [inline extension/chrome/elements/pgp_pubkey.ts]: Why is the `|| undefined` used?

File: extension/chrome/elements/pgp_pubkey.ts [hunk 2]
```diff
@@ -86,8 +89,8 @@ Catch.try(async () => {
       if (fixed !== armoredPubkey) { // try to re-render it after un-quoting, (minimized because it is probably their own pubkey quoted by the other guy)
         window.location.href = Env.urlCreate('pgp_pubkey.htm', { armoredPubkey: fixed, minimized: true, acctEmail, parentTabId, frameId });
       } else {
-        $('.line.add_contact').addClass('bad').text('This public key is invalid or has unknown format.');
-        $('.line.fingerprints').css({ display: 'none', visibility: 'hidden' });
+        $('#pgp_block.pgp_pubkey').empty()
+          .append('<div class="bad">This OpenPGP key is not usable.</div>'); // xss-direct
```
Comment (tomholub) [inline extension/chrome/elements/pgp_pubkey.ts]: Thanks, correct use of our xss comments :+1:

File: extension/chrome/elements/pgp_pubkey.ts [hunk 3]
```diff
@@ -98,7 +101,12 @@ Catch.try(async () => {
       for (const pubkey of pubs) {
         const email = Str.parseEmail(pubkey.users[0].userId ? pubkey.users[0].userId!.userid : '').email;
         if (email) {
-          contacts.push(await Store.dbContactObj({ email, client: 'pgp', pubkey: pubkey.armor(), lastUse: Date.now(), lastSig: await Pgp.key.lastSig(pubkey) }));
+          contacts.push(await Store.dbContactObj(
+            {
```
Comment (tomholub) [inline extension/chrome/elements/pgp_pubkey.ts]: I would usually put the `(` and `{` the same line as in:
```
doSomething({
  xxx, yyy, zzz,
  aaa, bbb, ccc
})
```

File: extension/chrome/elements/pgp_pubkey.ts [hunk 4]
```diff
@@ -98,7 +101,12 @@ Catch.try(async () => {
...
+              email, client: 'pgp', pubkey: pubkey.armor(), lastUse: Date.now(), lastSig: await Pgp.key.lastSig(pubkey),
+              expiresOn: Number(await Pgp.key.dateBeforeExpiration(pubkey)) || undefined
```
Comment (tomholub) [inline extension/chrome/elements/pgp_pubkey.ts]: `Number(await Pgp.key.dateBeforeExpiration(pubkey)) || undefined`
If this needs to be always done, we could (should) move it inside `dbContactObj`. So that here we can just do `expiresOn: await Pgp.key.dateBeforeExpiration(pubkey)`

File: extension/chrome/elements/pgp_pubkey.ts [hunk 5]
```diff
@@ -108,7 +116,8 @@ Catch.try(async () => {
     } else if (pubs.length) {
       if (Str.isEmailValid(String($('.input_email').val()))) {
         const contact = await Store.dbContactObj({
-          email: String($('.input_email').val()), client: 'pgp', pubkey: pubs[0].armor(), lastUse: Date.now(), lastSig: await Pgp.key.lastSig(pubs[0]),
+          email: String($('.input_email').val()), client: 'pgp', pubkey: pubs[0].armor(), lastUse: Date.now(),
+          lastSig: await Pgp.key.lastSig(pubs[0]), expiresOn: Number(await Pgp.key.dateBeforeExpiration(pubs[0])) || undefined
```
Comment (tomholub) [inline extension/chrome/elements/pgp_pubkey.ts]: Same here

File: extension/js/common/composer.ts [hunk 6]
```diff
@@ -1338,7 +1338,11 @@ export class Composer {
       for (const contact of renderableContacts) {
         ulHtml += `<li class="select_contact" data-test="action-select-contact" email="${Xss.escape(contact.email.replace(/<\/?b>/g, ''))}">`;
         if (contact.has_pgp) {
-          ulHtml += '<img src="/img/svgs/locked-icon-green.svg" />';
+          if ((contact.expiresOn || Number.MAX_SAFE_INTEGER) > Date.now()) {
```
Comment (tomholub) [inline extension/js/common/composer.ts]: Can use `|| Infinity`, a bit less technical 👍

File: extension/js/common/composer.ts [hunk 7]
```diff
@@ -1599,7 +1603,7 @@ export class Composer {
       this.debug(`renderPubkeyResult: Setting email to wrong / misspelled in harsh mode: ${email}`);
       $(emailEl).attr('title', 'This email address looks misspelled. Please try again.');
       $(emailEl).addClass("wrong");
-    } else if (contact.pubkey && await Pgp.key.usableButExpired((await openpgp.key.readArmored(contact.pubkey)).keys[0])) {
+    } else if (contact.pubkey && (contact.expiresOn || Number.MAX_SAFE_INTEGER) <= Date.now()) {
```
Comment (tomholub) [inline extension/js/common/composer.ts]: Here you will need to revert the change back, and still actually check the key.

Users will have keys in in the db that were imported before this change, an these keys don't have `exporesOn`, so this check would fail.

Maybe you can do both:
```js
contact.pubkey && ((contact.expiresOn || Number.MAX_SAFE_INTEGER) <= Date.now() || await Pgp.key.usableButExpired((await openpgp.key.readArmored(contact.pubkey)).keys[0]))
```
To skip the potentially expensive check at least sometimes.

File: extension/js/common/platform/store.ts [hunk 8]
```diff
@@ -242,10 +243,10 @@ export class Store {
   static getKeysCurrentlyInSession = async (acctEmail: string) => {
     const keys = await Store.keysGet(acctEmail);
     const result: Array<KeyInfo> = [];
-    for(const key of keys) {
+    for (const key of keys) {
       // Check if passpharse in the session
       if (!(await Store.passphraseGet(acctEmail, key.longid, true)) &&
-      (await Store.passphraseGet(acctEmail, key.longid, false))) {
+        (await Store.passphraseGet(acctEmail, key.longid, false))) {
```
Comment (tomholub) [inline extension/js/common/platform/store.ts]: awaits don't have to be bracketed, even if negated.

```js
if (! await Store.passphraseGet(acctEmail, key.longid, true) && await Store.passphraseGet(acctEmail, key.longid, false)) {
  // ...
}
```

**Comment by tomholub** [comment]
The test failure, I'm not sure. I suspect it may have something to do with the recent allow large attachments change, but I'm not sure.

**Comment by tomholub** [comment]
Yes, it does have to do with it. Because we enable trial on`test.ci.compose` for a short time as part of the test now (used to be a different email).

While trial is enabled, the extension wants user to sign into their account to take advantage of the added functionality. So it fails.

Looks like I will have to make a separate account for trial tests after all.

**Comment by tomholub** [comment]
The Gmail tests and the mock tests happen concurrently, therefore changing something on one account (on the backend) will affect both, because the backend is not mocked, it uses the live one.

File: extension/chrome/elements/pgp_pubkey.ts (hunk 1 above)
Comment (michael-volynets) [reply extension/chrome/elements/pgp_pubkey.ts]: oh, yes, I will remove it.

I needed it when I was saving `Boolean` to `Contact` model in our IndexedDB.
Thanks!

File: extension/js/common/composer.ts (hunk 7 above)
Comment (michael-volynets) [reply extension/js/common/composer.ts]: Yes, agree,
I forgot about that

**Comment by tomholub** [comment]
here
```js
      if (!pubkey) {
        return {
          email,
          name: name || null,
          pending_lookup: (pendingLookup ? 1 : 0),
          pubkey: null,
          has_pgp: 0, // number because we use it for sorting
          searchable: Store.dbCreateSearchIndexList(email, name || null, false),
          client: null,
          fingerprint: null,
          longid: null,
          longids: [],
          keywords: null,
          last_use: lastUse || null,
          pubkey_last_sig: null,
          pubkey_last_check: null,
          expiresOn: expiresOnMs || null
        };
      }
```
Since there is `!pubkey`, you can just `expiresOn: null`

**Comment by michael-volynets** [comment]
I missed that.
Thanks!

**Comment by tomholub** [comment]
Unusable key is still showing `...`, instead it should show more obvious message and remove the input and the button.

image

**Comment by tomholub** [comment]
Tested with this one:
```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: PGP Universal 3.3.2 (Build 21495)

mQINBFYKyAQBEADsjzxkjEm6ziOxJuPKw8DXyBL472vEM4B+u8JmfhL70CVLVQk9
el8shQcRXy0jBDSWE4ZOV1PMfZAVhgM9v1m3YS8bUBKbwYWehfW39x/+pjOnlu8W
KwN67on36Rn8uzQ6gWMVyfEl2pBGMm8kxGfgp2V41ZopZMPAe+gWJkV2I/IDdAfi
SOw8J0SY3d9yivNsDqC96iG2ckPRsZzrfpeAh/aiknBnzhhedmjV7mHP6LdyDGa2
AaxABkslmzZOnWjNGbTC0elHq//iaIq5Y2YS9zooKHSIJpY19QAxgnMVH1Aop3ml
0vmhIDcr7sFwXT2Qzu5OGA5YcpDH0GzkMtc9lOE3rU9l/P7R8nuQtkteKlCj9UCr
58N+I20XYAGskCjxY1hjhp3zSRMKNb8xWF+ShiImCfZlTkys1bCTdx/tjtXxKxSS
iquNwZeD0Wz5o9a2VOgnctTwk9IgdrUr7y++TNOFR8wjdZUw1ILbTu1yU2CJffVV
tzWfHvNDWt319+vtWXELz0weyCCbcIYj2v/xPfyV2DfZmRiAQY7sYXS3V/k4XjtL
UyzctVZ7YL/oK5xg+MImc5D0FDC8SBrX2SrNrngcRwoV+sFYX1PGpIVi0wxo8XKF
soSbzJpS+WLwZ+Kp5crhWO7S8BcGKp5KkYMRZgBkO+9SL8tYLooaH3Hj2QARAQAB
tB1KZWFuIFJPWSA8amVhbi5yb3lAdmFsZW8uY29tPokC2QQQAQIAwwUCVgrICQUJ
A8JnABsUAAAAAAARAAFrZXktdXNhZ2VAcGdwLmNvbY8gFAAAAAAAFgABa2V5LXVz
YWdlLW1hc2tAcGdwLmNvbYgwFIAAAAAAIAAHcHJlZmVycmVkLWVtYWlsLWVuY29k
aW5nQHBncC5jb21wZ3BtaW1lBgsJAwIBCgIZAQUXgAAAACMYbGRhcDovL2lzYzMt
c3YwMDYwMS52bmV0LnZhbGVvLmNvbQUbgwAAAAMWAgEFHgEAAAAEFQgJCgAKCRDh
wWmKxRnj5/bED/9KrAFjWVYYKI3HFfAmGQkziZt+FuKtdOvwtVCaIdsPcV+YbC4M
aTrL4eG6k3Hfxn/KtfStomXp26kULv1CKpeC1hy7ppXhTOOIqjlgCZ5iCsLVG4+6
sxzhEtoN2TiXUG7ddoI9z4FuMAI1AfV0DO3M1nsW5a+zKWn2b8QDoM9q9uvjOKMN
dy34tBW/+T3YOxLIrQytis2GyIPS+wEey6oK+AoqUtf15szma0pBDrZlO/Q2YmzT
q4IuGIFaXOK4vgXEAcd/sM18vFclx5hpM3gU3tw140foqzLu4BnRhPU0OycncMYQ
84FoS6VWyhPrQ2nL0aZRE6eZYRM5p4/2rI5M7NW9Y/zKGLD71Wpxtoje8OE+rIUA
6ECamxdlTcIS8AvJeCA2gMHYZv/iFfS+iLFtroN3J1zwLDvMr8I6QBzdEwwdzztI
RXHaXY7i5cu/w0zleG1tdyeBz+E+a6iDevfyvV1XSXp9CldC2gPO/t1H1RwerJ4F
ANDgjb0cc251WfGqjOcGijb+vznLOzYIm36QINAk2xCc3MihOl52BBLkBXAu6d56
gf5HKOS7u8NTFIOKfH34N2VuUEMzyv4zQ9+2kefghCe/SBMoz/09ecTWIRabS4sw
fVDaFZbCyZ92HsYgbCYLIHzWdESxcR8D0o1zm8c0Mk8lXX8OKcSYnshvoIkBIgQQ
AQIADAUCVgrIBgUDA8JnAAAKCRCoGdsnauJSe8MWB/9hakGaAFe8P9gqIiiDqDjS
dHt0Lm6iuqkSaDXKPh24KkQBiyjUJFyZSLNwDNTAzC/VyiRyiz7C/19wMlPdjKL8
iqwf7mZkrQsixI/WuG8+Q1NCxZVrl+6H8ozKzCCoVEMhSQYBZgSEjo/5Qd8wE/z0
p2yQLZjhXaM2WB6y6XCm34thUDyaFnC44Fkx1+8zt7yTBdolFc0TOtN3FKvLn6gk
8Ivnr2sgTwznKOcQKbl/h3RJvSsHJecF/pLSLU2cGDdzOuNDL9Hy8kv2h73dxlfw
z131e34N8KKLvP1RpTnKpjVQXS9GFmqpHyd207QcVfgSQ0PidLW/X+WgQpt/FTHI
tCdqZWFuLnJveUB2YWxlby5jb20gPGplYW4ucm95QHZhbGVvLmNvbT6JAtYEEAEC
AMAFAlYKyAYFCQPCZwAbFAAAAAAAEQABa2V5LXVzYWdlQHBncC5jb22PIBQAAAAA
ABYAAWtleS11c2FnZS1tYXNrQHBncC5jb22IMBSAAAAAACAAB3ByZWZlcnJlZC1l
bWFpbC1lbmNvZGluZ0BwZ3AuY29tcGdwbWltZQYLCQMCAQoFF4AAAAAjGGxkYXA6
Ly9pc2MzLXN2MDA2MDEudm5ldC52YWxlby5jb20FG4MAAAADFgIBBR4BAAAABBUI
CQoACgkQ4cFpisUZ4+f6Bw/+KPn8k3xs6M69tbXmY3Y+HneGymw/+BIRZx1GtBP4
ZvUzljDQsAc+i+Z414GQB6q4Mtaa6aqxBqAL/e/NJQkDfWzLxOb4sO/YDmMY1gi6
yhLDH/qfUsZCjGfmTUjXVyd4vbXwBI+Rctiihso0D9lMbfaqMvp5aENkx84S/jTB
hN5ejtDr4thoVdQTTX4XCC7gvYQR9Tini1Fd1KSow4HCnDcBiJxLuav2m1EkSoNe
Cnm7wvxnL8gQn5p74+I6B00xyZuFpBcR2glaJHHAcAHOHcS9JNA/aAVjzmJ++zQ6
Lw+JPvYCgiKR9lvvS4/wf5pvDARSSje++2iJg7boeRyhvMatqFLkgaSbNBC29N3g
1XjU3V2U22Y53nuC2AaZ+AHAYmYoIh31mPqagEXkfyEDaYPpw68oUyUaD9Jtpgqa
nSXNO95yRzGCwIRgv8pG9tFjmE22wMvG6OeC6AU5tlwiLWNdFGt1ndjhlDxiYHRp
cKy4L8r2J1AQMiIAFugw9B3/beVVc6Mo7j7Vj+Oay/l9CcNyt9ZaZXiV27MgMU8h
MndKnc580CvPKXj0UG9QgehA43aJFluq+sjXVRYYcV/XYvUli8bL7nBgQyKyPiHI
jjaXe2IFvb/R/ej6fNALua0XvpiqgGInE1AuFSuFJW306sWzQuX3kr1GJX/j/ilC
JOyJASIEEAECAAwFAlYKz0IFAwPCZwAACgkQqBnbJ2riUnvkpAf+LAnrhBWv9lls
kjwZr1FLoEcQQ7MZIcIN4QXIQfoAMkDQ8cGhn1qDbd+ABkAAXEks30JylJXYQlwd
+uSuAa9Y86QqUJ8dj12kF3VpORp+uxGS7zhs1+81JAZ2sVf5xtnCg4shxkunDpKC
y9IVKqssaneMCswdfcpT1ZtXyoNtHJ65BhQJqHxIGgSs5gowZ6P+O+FlV4o3nQ7h
Ty/YD+PSyh6GHAmb0rggqEwvpwSzOBViRXTjKYE2M3QhDy+w/wiV4wGJIRsANecU
87aginJzfEfOoujaSHGeJofpxXeaW6vNaTf/0IeMv3hpNdcuQ4QBbh26/Q0IKPT0
V6htSoLstYkGRwQQZAIGOgUCWc2ANAUDA8JnAAIbAsVpZAEBBDCCBiEwggUJoAMC
AQICFGhqNxsrDdpK5TNn6OHBaYrFGePnMA0GCSqGSIb3DQEBBQUAMIGQMRIwEAYD
VQQDEwl2YWxlby5jb20xDjAMBgNVBAoTBVZhbGVvMQ4wDAYDVQQLEwVJU1NDQzEP
MA0GA1UEBxMGRnJhbmNlMRQwEgYDVQQIEwtMYSBWZXJyaWVyZTELMAkGA1UEBhMC
RlIxJjAkBgkqhkiG9w0BCQEWF2lzc2NjLm1haWxib3hAdmFsZW8uY29tMB4XDTE3
MDkyODIzMDUyNFoXDTE5MDkyODIzMDUyNFowgZQxCzAJBgNVBAYTAkZSMQ8wDQYD
VQQIEwZGcmFuY2UxFDASBgNVBAcTC0xhIFZlcnJpZXJlMQ4wDAYDVQQLEwVJU1ND
QzEOMAwGA1UEChMFVmFsZW8xITAfBgkqhkiG9w0BCQEWEmplYW4ucm95QHZhbGVv
LmNvbTEbMBkGA1UEAwwSamVhbi5yb3lAdmFsZW8uY29tMIICIjANBgkqhkiG9w0B
AQEFAAOCAg8AMIICCgKCAgEA7I88ZIxJus4jsSbjysPA18gS+O9rxDOAfrvCZn4S
+9AlS1UJPXpfLIUHEV8tIwQ0lhOGTldTzH2QFYYDPb9Zt2EvG1ASm8GFnoX1t/cf
/qYzp5bvFisDeu6J9+kZ/Ls0OoFjFcnxJdqQRjJvJMRn4KdleNWaKWTDwHvoFiZF
diPyA3QH4kjsPCdEmN3fcorzbA6gveohtnJD0bGc636XgIf2opJwZ84YXnZo1e5h
z+i3cgxmtgGsQAZLJZs2Tp1ozRm0wtHpR6v/4miKuWNmEvc6KCh0iCaWNfUAMYJz
FR9QKKd5pdL5oSA3K+7BcF09kM7uThgOWHKQx9Bs5DLXPZThN61PZfz+0fJ7kLZL
XipQo/VAq+fDfiNtF2ABrJAo8WNYY4ad80kTCjW/MVhfkoYiJgn2ZU5MrNWwk3cf
7Y7V8SsUkoqrjcGXg9Fs+aPWtlToJ3LU8JPSIHa1K+8vvkzThUfMI3WVMNSC207t
clNgiX31Vbc1nx7zQ1rd9ffr7VlxC89MHsggm3CGI9r/8T38ldg32ZkYgEGO7GF0
t1f5OF47S1Ms3LVWe2C/6CucYPjCJnOQ9BQwvEga19kqza54HEcKFfrBWF9TxqSF
YtMMaPFyhbKEm8yaUvli8GfiqeXK4Vju0vAXBiqeSpGDEWYAZDvvUi/LWC6KGh9x
49kCAwEAAaOCAWswggFnMB0GCisGAQQBmkkIAQEEDxcNMTUwOTI5MTcxOTAwWjAO
BgNVHQ8BAf8EBAMCB4AwEwYDVR0lBAwwCgYIKwYBBQUHAwQwEQYDVR0OBAoECOHB
aYrFGePnMDEGA1UdEQQqMCiBEmplYW4ucm95QHZhbGVvLmNvbYESamVhbi5yb3lA
dmFsZW8uY29tMBMGA1UdIwQMMAqACKgZ2ydq4lJ7MIHFBgNVHR8Egb0wgbowgbeg
gbSggbGGNGh0dHA6Ly9rZXlzLnZhbGVvLmNvbTo4MC9jcmwvUmV2b2tlZENlcnRp
ZmljYXRlcy5jcmyGeWxkYXA6Ly9rZXlzLnZhbGVvLmNvbTozODkvY249UmV2b2tl
ZCUyMENlcnRpZmljYXRlcyxvPUNSTD9jZXJ0aWZpY2F0ZVJldm9jYXRpb25MaXN0
P2Jhc2U/b2JqZWN0Y2xhc3M9Y1JMRGlzdHJpYnV0aW9uUG9pbnQwDQYJKoZIhvcN
AQEFBQADggEBAJiy7oxnjFuyfNQn+R+M28raXSbDxG9Xfp46tZ6AjJL/wfgieYAH
oFM8TiZn3Z/6a0ZsOs7Eqd7gsfwTr9YPbjd+htgfW+7YC+1N2SKC/gvOQX89azAP
iMj0MkAtZVQW/Dkyo8Mf1LTJtq+aAsThl/2HKIbWtYOe63jgB+Fa9I+TnaK+rOCZ
IdkhR+HuiqV4MqUml8B2wCb3n9RtBc8xT+kDAW597LRcmSgki8Z12S9tgEgwlNcb
rA8be5JJ8pao8ymZT+a8usLnBMV2SMIEgNLz9/zX9VMKHkoKDbP4fQs6ndHRIOtk
AkxYDxB9qeSmDG9odOJhCWsqRfpWrjryjhYAAAAAAAEBuQINBFYKyAYBEADyADkw
yEXOd4Wg+JzkC3HeJhBt9sTFCFbPmG8+ivFP/X1drPkjXkcHaWhIFj/WMTQheQdO
CJclU1SsgP6knUkAj59Yh4oePG6x/I3lFcS+kgefwFffHTkHh4h2yRRTp57LU+Ck
U1k+aFF5As3DlGIm5JC+0+6f6dJ0E7tDDqHQ6HjmO4HQ8/mdJWm/xWQdaHIWODxs
zlw8ZYKRGNXdwK7bWCRrCVGeKDrdpywyIg3AKIcZZ6r0C8n12xHmY7nOekZwYr3c
73woeqMX49ca8ZKqJ/caRxNI2vaKcuKNs+719cyOgadnm6KSgtKDfnB+t/1fuKNM
rrxFkXJ20YPy2sFHtTwVlfjXQYj4v9fPgQD65xZeZ2Vk8QWiwr8/sxBurtrNrqZB
linkPrX0F1h0ep0qFuXbnNrlsDacu3A+Bk9/IOQaijngMr3ASG+GxRRQ6uMyUWjH
aHV33QKAkRbOYTbCGW1juIrxhDMNPxcdePN5RGOgTmEn9mOpr0lEwTZyJ+AB7ag0
HJ+SZjErBNsRq49Cc3Sr+Y1vvWgTdX+GzfLnPI1QPSnmWt+U3btmao9D6r0pH6EV
XJk9dll59Z259Mr0gw1uv73c0YXrbSqsqFDGDH/4EyJTslwbbrlOnRNdzAa23deq
Dmd9xb+Peyu8ZOlcvsx3p86ndOFWfvZybclUMQARAQABiQqOBBgBAgh4BQJZzYA0
BQkHhcfuxYUUAAAAAAAXBiV4NTA5Y2VydGlmaWNhdGVAcGdwLmNvbTCCBiEwggUJ
oAMCAQICFHHfsnDJjLkBY/ay67lzGhSdSEbWMA0GCSqGSIb3DQEBBQUAMIGQMRIw
EAYDVQQDEwl2YWxlby5jb20xDjAMBgNVBAoTBVZhbGVvMQ4wDAYDVQQLEwVJU1ND
QzEPMA0GA1UEBxMGRnJhbmNlMRQwEgYDVQQIEwtMYSBWZXJyaWVyZTELMAkGA1UE
BhMCRlIxJjAkBgkqhkiG9w0BCQEWF2lzc2NjLm1haWxib3hAdmFsZW8uY29tMB4X
DTE3MDkyODIzMDUyNFoXDTE5MDkyODIzMDUyNFowgZQxCzAJBgNVBAYTAkZSMQ8w
DQYDVQQIEwZGcmFuY2UxFDASBgNVBAcTC0xhIFZlcnJpZXJlMQ4wDAYDVQQLEwVJ
U1NDQzEOMAwGA1UEChMFVmFsZW8xITAfBgkqhkiG9w0BCQEWEmplYW4ucm95QHZh
bGVvLmNvbTEbMBkGA1UEAwwSamVhbi5yb3lAdmFsZW8uY29tMIICIjANBgkqhkiG
9w0BAQEFAAOCAg8AMIICCgKCAgEA8gA5MMhFzneFoPic5Atx3iYQbfbExQhWz5hv
PorxT/19Xaz5I15HB2loSBY/1jE0IXkHTgiXJVNUrID+pJ1JAI+fWIeKHjxusfyN
5RXEvpIHn8BX3x05B4eIdskUU6eey1PgpFNZPmhReQLNw5RiJuSQvtPun+nSdBO7
Qw6h0Oh45juB0PP5nSVpv8VkHWhyFjg8bM5cPGWCkRjV3cCu21gkawlRnig63acs
MiINwCiHGWeq9AvJ9dsR5mO5znpGcGK93O98KHqjF+PXGvGSqif3GkcTSNr2inLi
jbPu9fXMjoGnZ5uikoLSg35wfrf9X7ijTK68RZFydtGD8trBR7U8FZX410GI+L/X
z4EA+ucWXmdlZPEFosK/P7MQbq7aza6mQZYp5D619BdYdHqdKhbl25za5bA2nLtw
PgZPfyDkGoo54DK9wEhvhsUUUOrjMlFox2h1d90CgJEWzmE2whltY7iK8YQzDT8X
HXjzeURjoE5hJ/Zjqa9JRME2cifgAe2oNByfkmYxKwTbEauPQnN0q/mNb71oE3V/
hs3y5zyNUD0p5lrflN27ZmqPQ+q9KR+hFVyZPXZZefWdufTK9IMNbr+93NGF620q
rKhQxgx/+BMiU7JcG265Tp0TXcwGtt3Xqg5nfcW/j3srvGTpXL7Md6fOp3ThVn72
cm3JVDECAwEAAaOCAWswggFnMB0GCisGAQQBmkkIAQEEDxcNMTUwOTI5MTcxOTAy
WjAOBgNVHQ8BAf8EBAMCAzgwEwYDVR0lBAwwCgYIKwYBBQUHAwQwEQYDVR0OBAoE
CLlzGhSdSEbWMDEGA1UdEQQqMCiBEmplYW4ucm95QHZhbGVvLmNvbYESamVhbi5y
b3lAdmFsZW8uY29tMBMGA1UdIwQMMAqACKgZ2ydq4lJ7MIHFBgNVHR8Egb0wgbow
gbeggbSggbGGNGh0dHA6Ly9rZXlzLnZhbGVvLmNvbTo4MC9jcmwvUmV2b2tlZENl
cnRpZmljYXRlcy5jcmyGeWxkYXA6Ly9rZXlzLnZhbGVvLmNvbTozODkvY249UmV2
b2tlZCUyMENlcnRpZmljYXRlcyxvPUNSTD9jZXJ0aWZpY2F0ZVJldm9jYXRpb25M
aXN0P2Jhc2U/b2JqZWN0Y2xhc3M9Y1JMRGlzdHJpYnV0aW9uUG9pbnQwDQYJKoZI
hvcNAQEFBQADggEBAEmDSZ5Sf0pa/Tg5RnMft/9DZ1e0TRkBIgYRF70aNFCTVoEb
3UIq66/BVoyx5rR69OUcxlgBDmh8K6BqLdGPDBfx3+Ry15THG6PTFKM/8LXpKD7s
ybGqCUF1uN/ngnuGNys0+pdAOhnSVLTaqfaKjtZOvGbv8HZcFJXWk9yrin5HMUM5
+z83PZQmc6XyE7ygtunjiOo4noobphnwjDupHQAafLlXo0so5Z/Q2JH6CdCny4xv
TvoJ9tUGkpsjbsv8BkoH+0/NopIXEI/QS8m+k+sXM9trwzjQVQi0CwwUZ9mvSJS/
+ADcyVc01PwGanSHAU09PO4yHUxrJaLlfNnp0REFGwwAAADBXSAEGQEIAAYFAlYK
yAYACgkQuXMaFJ1IRtaw2g//bQTvk3oMToj0HqaYr4ptWqSy7eaO3rkir8eSXdz6
hydj5wU1fdfAYS16dbqBbszvBX6pDOuldoBD+q3yIJrDJ0cwH1UGodgQeHK79rsL
THr2javbQLDbsRUNGxKG7vHJ2JpwNCs3s0ymohZJTpgckr5vnoSOX5wXvuGutnJM
kg6XwgAZFoC0E/Y/Bfxblb8Nq8Qzk2Bhn8Vz6cuOURILb4s8jB/yvtxkgKOFsVWC
krtP6HL80XYSWuqvZQLVLkNztrBc5BtlFDJms6IR5W2R4ix0jgsmgemiqFzcb6G0
BipG6+g32ngOq31KDd+2ab3b5yEy6t3vLJUUIdqmdQSOfLGSGbdD2TtH4HZRx0iS
4pJU1Xkd64uobNUY4ZnVeFH/9gvt4YbUeBHYngkfqp93bS4sR6i2ZeAV/4m3hOnN
0n7D9WwbtsijeNkgQumT3YHRaebsuVlmOMI70Qpy4RkFDcLjOJP2CEx2B/qmPDbq
/AEh5ebiwj8V84iNWMJAY+qGVs9raVgNi6a91h/4bj4BNgsxR/khojDxsw1FeGbe
p42kf+SDuQf4tDbOrteFo/D8bgBaaVUk+ZoGioeHYIJ3yhLDvWig/zwUMnvv98Jq
vsObWneVr3R2/BOeM5bABs3pmhREXgXkmBgcJCfskG8UbAunrRP2rr020QqCUC5R
65oACgkQ4cFpisUZ4+fVyhAAh68Jhnpq2bQKcAOqaneI33ts3SOY5N056rFL3L9z
nfPo2u3+dAJc5TtaT/oF3exMn63iNCm0OkuqncwuQ8002AcGiGL7O8QhzAkQNr6f
fXNDXvrNFvZnfgYhGwaPkhaGo4CaMvpRtVj3ZpjN8lT/Pa65wmdGm9iWdQYAAskn
e47MO1/5CDSOUOgDalBZQ5PuDJ+FAH5ihGMxjqa+vReDSpopIWMgSTpOmCzG0yk+
XSCtyrNFeDoxSIgfVQ4rOaOQGXCwwskQulTK46edFfWMWSN9mSbqzRUjzOCO7gn1
LGFwroL22Osr5YuuOLsG6K5mk1ouNiyumAyQ9qB2ltq1gD1p3jwFsHqp71WH7Nr8
g9QUSlscGH77wCsYZ68P2eULASSbcvka4VtLQCtq1r/udYuWuU/7SrNlWDV/KupG
HHPwHMWSNnPFH5inGTx8B7/0Li4AdxmusmyRqxrgE0vqrBV/yzVtmI0xeF83tfTo
RAmnqnf+FBEp3gYO9FsUq1ovQjs5E1YTp4c/YXcDEE/owEnS/XHrU4DruehMHayl
dBie+BNHyot1sHiSZ1VZH1KvAU1rKZ6/MxSeGEXKIu0Q6XQ1V7mWJwbFrFlah7mX
UyS+o4GC9czYEhZonjTXAd65B+jQaOmIjENAMu5ECYzjjhk9XLib6m3oyN+AuyRC
ZmI=
=PCtZ
-----END PGP PUBLIC KEY BLOCK-----
```

**Comment by michael-volynets** [comment]
Sorry, I think I missed that, let me check again

**Comment by tomholub** [comment]
This looks much better! The previous expired design was really ugly.

image

**Comment by michael-volynets** [comment]
The tests fail because it throws an error that says this:
image

How can I renew these settings?

**Comment by tomholub** [comment]
on the compose account?

**Comment by tomholub** [comment]
I'll reset the account on backend

**Comment by michael-volynets** [comment]
Yes, Thanks!

**Comment by tomholub** [comment]
Ok now should work you can retry it

image
